### PR TITLE
Fix macOS support

### PR DESCRIPTION
**Pull request by yanyin1986**
ref: https://github.com/kean/NukeUI/pull/46

NukeUI v0.8.2 can't go well on the macOS.
And I fixed that.

### CODE REVIEW

File: Sources/VideoPlayerView.swift [hunk 1]
```diff
@@ -92,31 +92,35 @@ public final class VideoPlayerView: _PlatformBaseView {
...
     }
-    
+
     private func registerNotifications() {
         NotificationCenter.default
             .addObserver(self,
                          selector: #selector(registerNotification(_:)),
                          name: .AVPlayerItemDidPlayToEndTime,
                          object: player?.currentItem)
+        #if os(iOS) || os(tvOS)
```
Comment (yanyin1986) [inline Sources/VideoPlayerView.swift:102]: According to https://developer.apple.com/documentation/uikit/uiapplication/1622944-willenterforegroundnotification

`UIApplication.willEnterForegroundNotification` seems only works on iOS (macCatalyst) and tvOS.

**Comment by kean** [comment]
Thanks! NukeUI is moving to the main repo with [Nuke 11](https://github.com/kean/Nuke/pull/547), so it now has a complete set of unit tests. I'll merge your changes now and release a patch.

**Comment by kean** [comment]
Patch https://github.com/kean/NukeUI/releases/tag/0.8.3

**Comment by yanyin1986** [comment]
Thanks! 😁

**Pull request merged**